lib/module-types.js

```diff
@@ -0,0 +1,75 @@
+const requireTrueValue = value => {
+	if (value !== true) {
+		throw new TypeError('When specifying module types, use `true` for ’cjs’, ’mjs’ and ’js’ extensions');
+	}
+};
+
+const normalize = (extension, type, defaultModuleType) => {
+	switch (extension) {
+		case 'cjs':
+			requireTrueValue(type);
+			return 'commonjs';
+		case 'mjs':
+			requireTrueValue(type);
+			return 'module';
+		case 'js':
+			requireTrueValue(type);
+			return defaultModuleType;
+		default:
+			if (type !== 'commonjs' && type !== 'module') {
+				throw new TypeError(`Module type for ’${extension}’ must be ’commonjs’ or ’module’`);
+			}
+
+			return type;
+	}
+};
+
+const deriveFromObject = (extensionsObject, defaultModuleType) => {
+	const moduleTypes = {};
+	for (const [extension, type] of Object.entries(extensionsObject)) {
+		moduleTypes[extension] = normalize(extension, type, defaultModuleType);
+	}
+
+	return moduleTypes;
+};
+
+const deriveFromArray = (extensions, defaultModuleType) => {
+	const moduleTypes = {};
+	for (const extension of extensions) {
+		switch (extension) {
+			case 'cjs':
+				moduleTypes.cjs = 'commonjs';
+				break;
+			case 'mjs':
+				moduleTypes.mjs = 'module';
+				break;
+			case 'js':
+				moduleTypes.js = defaultModuleType;
+				break;
+			default:
+				moduleTypes[extension] = 'commonjs';
```

Comment:
What if this `default` behaviour was an option @novemberborn?

Defaults to `'commonjs'` — keeping backwards compatibility — but also configurable as `'module'`.